### PR TITLE
chore(iast): fastapi query params support

### PR DESCRIPTION
**Pull request by avara1986**
Taint the query string source for IAST integration with FastAPI

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/appsec/_iast/_patch.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
+import functools
 import sys
 from typing import Callable
 from typing import Text
 
 from wrapt import FunctionWrapper
+from wrapt import wrap_function_wrapper as _w
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_patch.py:7]: Why not use our own `try_wrap_function_wrapper` defined in line 37 of this file?

It's the same function but with a try/except for unexpected error handling

File: tests/contrib/fastapi/test_fastapi_appsec_iast.py
Comment (gnufede) [inline tests/contrib/fastapi/test_fastapi_appsec_iast.py:1]: Maybe add an entry for this file to the CODEOWNERS?

**Review by gnufede — APPROVED** [review]

File: tests/contrib/fastapi/test_fastapi_appsec_iast.py
Comment (christophe-papazian) [reply tests/contrib/fastapi/test_fastapi_appsec_iast.py:1]: It should not be necessary as all test files with appsec in their name are asm owned.
https://github.com/DataDog/dd-trace-py/blob/main/.github/CODEOWNERS#L106

File: ddtrace/appsec/_iast/_patch.py (hunk 1 above)
Comment (avara1986) [reply ddtrace/appsec/_iast/_patch.py:7]: :+1:

File: tests/contrib/fastapi/test_fastapi_appsec_iast.py
Comment (avara1986) [reply tests/contrib/fastapi/test_fastapi_appsec_iast.py:1]: This file is covered by this rule:

```
tests/contrib/*/test*appsec*.py     @DataDog/asm-python
```

**Review by christophe-papazian — APPROVED** [review]

**Review by juanjux — APPROVED** [review]

File: ddtrace/contrib/internal/fastapi/patch.py [hunk 2]
```diff
@@ -74,14 +75,15 @@ def patch():
...
         _w("fastapi", "BackgroundTasks.add_task", _trace_background_tasks(fastapi))
-
     # We need to check that Starlette instrumentation hasn't already patched these
     if not isinstance(fastapi.routing.APIRoute.handle, ObjectProxy):
         _w("fastapi.routing", "APIRoute.handle", traced_handler)
 
     if not isinstance(fastapi.routing.Mount.handle, ObjectProxy):
         _w("starlette.routing", "Mount.handle", traced_handler)
 
+    _on_iast_fastapi_patch()
```
Comment (brettlangdon) [inline ddtrace/contrib/internal/fastapi/patch.py:85]: would it be better to have something like:

```python
if iast_is_enabled():
    from ddtrace.appsec._iast._patch import _on_iast_fastapi_patch
    _on_iast_fastapi_patch()
```

Are we concerned about any side effects from importing `iast` when it isn't going to be used?

**Review by brettlangdon — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Pull request merged**